Mastodon/Scene/Thread/Edit History/StatusEditHistoryTableViewCell.swift

```diff
@@ -0,0 +1,134 @@
+        grayBackground.layer.borderColor = Asset.Scene.EditHistory.statusBackgroundBorder.color.cgColor
+        grayBackground.applyCornerRadius(radius: 8)
+
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+        selectionStyle = .none
+        grayBackground.addSubview(statusHistoryView)
+        contentView.addSubview(dateLabel)
+        contentView.addSubview(grayBackground)
+        
+        setupContainerViewMarginConstraints()
+        setupConstraints()
+        updateContainerViewMarginConstraints()
+    }
+
+    required init?(coder: NSCoder) { fatalError("init(coder:) has not been implemented") }
+
+    private func setupConstraints() {
+        statusViewBottomConstraint = statusHistoryView.bottomAnchor.constraint(equalTo: grayBackground.bottomAnchor, constant: -Self.verticalMargin)
+        let constraints = [
+            dateLabel.topAnchor.constraint(equalTo: contentView.topAnchor, constant: 8),
+            dateLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
+            dateLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
+
+            grayBackground.topAnchor.constraint(equalTo: dateLabel.bottomAnchor, constant: Self.verticalMargin),
+            grayBackground.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
+            grayBackground.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
+            grayBackground.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -Self.verticalMargin),
+
+            statusHistoryView.topAnchor.constraint(equalTo: grayBackground.topAnchor, constant: Self.verticalMargin),
+            statusHistoryView.leadingAnchor.constraint(equalTo: grayBackground.leadingAnchor),
+            statusHistoryView.trailingAnchor.constraint(equalTo: grayBackground.trailingAnchor),
+            statusViewBottomConstraint,
+        ].compactMap { $0 }
+
+        NSLayoutConstraint.activate(constraints)
+    }
+
+    func configure(status: Status, statusEdit: StatusEdit, dateText: String) {
+        dateLabel.text = dateText
+        statusHistoryView.statusView.configure(status: status, statusEdit: statusEdit)
+    }
+    
+    override func prepareForReuse() {
+        statusHistoryView.prepareForReuse()
+        super.prepareForReuse()
+    }
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        updateContainerViewMarginConstraints()
+    }
+}
+
+// MARK: - AdaptiveContainerMarginTableViewCell
+extension StatusEditHistoryTableViewCell: AdaptiveContainerMarginTableViewCell {
+    var containerView: StatusHistoryView {
+        statusHistoryView
+    }
+}
+
+class StatusHistoryView: UIView {
+    let statusView = StatusView()
+    
+    private var statusViewLeadingConstraint: NSLayoutConstraint!
```

Comment:
The reason for force unwrapping (`NSLayoutConstraint!`) is that you can only set the constraints after adding `statusView` as a subview in `init`, right? What's your opinion on `NSLayoutConstraint?` instead?

Reply:
Exactly, but we can easily remove the implicit unwrap, I've done this now 👍